### ISSUE TITLE
Feature Request: show images on demand.

### ISSUE THREAD

**Issue by Eskibear** [NessajCN/opens1#5]
It would extremely improve user experience if we can view images (on demand, of course).

By parsing `img` tag under each post node, it's feasible to learn url of images. 
![image](https://github.com/NessajCN/opens1/assets/2351748/aa8768ed-ff4c-441a-b1ae-7d61a4b0ac87)

I can think of below approaches:
- a) add image urls as part of the message directly, so we have chance to open them if we want. Simple and effective, unblocking the use case.
- b) render as a particular pattern containing the url, and register a hover provider implementing a customized webview rendering the image. Pro: we can hover on the content to view image, which is convenient and fancy. Cons: more effort, you need to be careful not to deteriorate performance, caching images if necessary to avoid repeated requests, etc.
- or other visualization...

Personally I prefer option a), which look enough for the moment.

**Comment by NessajCN** [comment]
It's been considered to add image view. The implementation was fairly simple though I canceled the plan. To display images I have to open threads in webview which consumes much higher resources and there's no point to open a fully-functioned webpage in vscode ( "why do I not just use a browser"). Say it's just for slacking off at work XD. Anyway thanks for your suggestion.

**Comment by Eskibear** [comment]
Appreciate the quick response. Agree that we don’t render images for
performance concern.

Like I mentioned above, can we simply include image url in the message if
exists? It unblocks the use case that anyone who wants to view image(copy
paste and open in browser), while not making big trouble to other users.

But indeed it’s a nice to have feature and I can live without it. Anyway
thank you for your effort and the extension is excellent.
-- 
- Yan Zhang
